Core/GDCore/Project/EventsFunctionsExtension.cpp

```diff
@@ -138,11 +144,40 @@ void EventsFunctionsExtension::UnserializeFrom(
     dependencies.push_back(
         UnserializeDependencyFrom(dependenciesElement.GetChild(i)));
 
+  // Only unserialize behaviors and objects names.
+  // As event based objects can contains CustomObject and Custom Object,
+  // this allows them to reference EventBasedBehavior and EventBasedObject
+  // respectively.
+  auto &behaviorsElement = element.GetChild("eventsBasedBehaviors");
+  behaviorsElement.ConsiderAsArrayOf("eventsBasedBehavior");
+  for (std::size_t i = 0; i < behaviorsElement.GetChildrenCount(); ++i) {
+    const gd::String &behaviorName =
+        behaviorsElement.GetChild(i).GetStringAttribute("name");
+    eventsBasedBehaviors.InsertNew(behaviorName, eventsBasedBehaviors.GetCount());
+  }
+  auto &objectsElement = element.GetChild("eventsBasedObjects");
+  objectsElement.ConsiderAsArrayOf("eventsBasedObject");
+  for (std::size_t i = 0; i < objectsElement.GetChildrenCount(); ++i) {
+    const gd::String &objectName =
+        objectsElement.GetChild(i).GetStringAttribute("name");
+    eventsBasedObjects.InsertNew(objectName, eventsBasedObjects.GetCount());
+  }
+}
+
+void EventsFunctionsExtension::UnserializeExtensionImplementationFrom(
+    gd::Project& project,
+    const SerializerElement& element) { 
   UnserializeEventsFunctionsFrom(project, element.GetChild("eventsFunctions"));
-  eventsBasedBehaviors.UnserializeElementsFrom(
-      "eventsBasedBehavior", project, element.GetChild("eventsBasedBehaviors"));
-  eventsBasedObjects.UnserializeElementsFrom(
-      "eventsBasedObject", project, element.GetChild("eventsBasedObjects"));
+  auto &behaviorElement = element.GetChild("eventsBasedBehaviors");
```

Comment:
There is a risk the lists were manipulated between the calls to UnserializeExtensionDeclarationFrom and UnserializeExtensionImplementationFrom.
I think this needs at least a check that the lists length are the same, and a check that every name is still the same.
Otherwise, this should probably error (bail out before unserializing if these verifications are not ok).